samples/app-anonymous-users/nodejs/ClientApp/src/components/shareview.jsx

```diff
@@ -0,0 +1,320 @@
+
+    // Declare new state variables that are required for a verified anonymous user or a normal user
+    const [IsAnonymousUser, setIsAnonymousUser] = useState(false);
+
+    // Declare new state variables that are required for a verified anonymous user or a normal user
+    const [IsConsentButtonVisible, setIsConsentButtonVisible] = useState(false);
+
+    // Declare new state variables that are required disable the authentication button after login
+    const [ssoAuthenticationButtonVisible, setIsssoAuthenticationButtonVisible] = useState(true);
+
+    useEffect(() => {
+        microsoftTeams.app.initialize();
+        verifyAnonymousUser();
+    }, [])
+
+    // Builds the socket connection, mapping it to /io
+    useEffect(() => {
+        microsoftTeams.app.initialize();
+        setSocket(io());     
+    }, []);
+
+     // subscribe to the socket event
+     useEffect(() => {
+        if (!socket) return;
+        socket.on('connection', () => {
+            socket.connect();
+        });
+
+       // receive a message from the server
+        socket.on("message", data => {
+            if (data.Key === "Anonymous") {
+                aShowSetCount(data.Value);
+            } else {
+                uShowSetCount(data.Value);
+            }
+        });
+    
+    }, [socket]);
+  
+    // Once we call getContext API, we can recognize anonymous users by checking for the licenseType value like: context.user.licenseType === "Anonymous".
+    const verifyAnonymousUser = () => {
+          microsoftTeams.app.getContext().then((context) => {
+            if (context.user.licenseType === "Anonymous") {
+                setIsAnonymousUser(true);
+            }
+        });
+    }
+
+    // Click submit vote button
+    const submitVote = () => {
+
+        if (socket) {
+            setIsVoteBtnDisabled(true); // Disable Button
+            if (IsAnonymousUser) {
+                // Update the state property for incremented count value.
+                let addAnonymousVal = aShowCount + 1;
+
+                // Sending the updated count to socket.emit to show the latest data on stage view.
+                var anonymousCountValue={  
+                    Key : "Anonymous",  
+                    Value : addAnonymousVal  
+                    };
+                socket.emit("message", anonymousCountValue); // send a message to the server
+            }
+            else {
+                // Update the state property for incremented count value.
+                let addUserVal = uShowCount + 1;
+
+                // Sending the updated count to socket.emit to show the latest data on stage view.
+                var UserCountValue={  
+                    Key : "User",  
+                    Value : addUserVal  
+                    };
+                socket.emit("message", UserCountValue); // send a message to the server
+            }
+        }
+        else {
+            alert('No connection to server yet.');
```

Comment:
Do we need this alert

Reply:
No, but because the socket is not connected, we must display an alert. 